archive/tar_test.ts

```diff
@@ -88,7 +88,7 @@ Deno.test("Tar() appends file with long name to tar archive", async function ():
   assert(result !== null);
   assert(!result.consumed);
   const untarText = new TextDecoder("utf-8").decode(await readAll(result));
-  assert(result.consumed);
+  // assert(result.consumed);
```

Comment:
When enabled, this fails because `TarEntry.read()` doesn't change `TarEntry.#consumed` to `true` once fully read. I'm not sure why yet.

Reply:
maybe we don't need to touch `std/archive` in this PR as it's re-designed completely in #1985?

Reply:
We'll have to fix this if we're going to merge this PR before legacy `std/archive` is removed.

Reply:
Fixed.